prody/dynamics/plotting.py

```diff
@@ -2381,3 +2355,43 @@ def showTree_networkx(tree, node_size=20, node_color='red', node_shape='o',
         showFigure()
 
     return mpl.gca()
+
+
+def checkColors(colors, num, labels, allowNumbers=False):
+    """Check colors and process them if needed"""
+
+    from matplotlib.colors import is_color_like
+
+    colors_dict = {}
+
+    if isinstance(colors, np.ndarray):
+        colors = tuple(colors)
+
+    if is_color_like(colors) or colors is None:
+        colors = [colors] * num
+    elif isListLike(colors):
+        colors = list(colors)
+
+    if isinstance(colors, dict):
```

Comment:
Should this be an `elif` (for better readability)?

Reply:
Good point. I was thinking that this is a separate thing, but it can't be true at the same time as the others, so yes.